core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -211,7 +211,7 @@ public ManagedChannelImpl build() {
         target,
         // TODO(carl-mastrangelo): Allow clients to pass this in
         new ExponentialBackoffPolicy.Provider(),
-        firstNonNull(nameResolverFactory, NameResolverRegistry.getDefaultRegistry()),
+        firstNonNull(nameResolverFactory, NameResolverProvider.asFactory()),
```

Comment:
Update the comments on `ManagedChannelBuilder.nameResolverFactory()` to mention that the service providers will be used by default.